### ISSUE TITLE
[RFE] compare two reports generated by preupg

### ISSUE THREAD

**Issue by phracek** [upgrades-migrations/preupgrade-assistant#158]
Preupg assessments system and generates report in XML file.

Either new binary or new option in preupg called `--diff` should be able to compare two or more reports and provides differences. 
Be carefull XML file contains some garbage which should be filter out.

Example XML file:

```
<ns0:Benchmark id="xccdf_preupg-content_benchmark_all" resolved="1" xmlns:xhtml="http://www.w3.org/1999/xhtml/" xml:lang="en" xmlns:ns0="http://checklists.nist.gov/xccdf/1.2">
!!!!!!!  <ns0:status date="2016-10-12">draft</ns0:status> !!!!!!!
  <ns0:title xml:lang="en">Preupgrade Assistant migration tool</ns0:title>
  <ns0:description xml:lang="en">
    <html:p xmlns:html="http://www.w3.org/1999/xhtml">
      This is a checklist of configuration settings intended to be used as RedHat Enterprise Linux migration tool.
    </html:p>
    <html:p xmlns:html="http://www.w3.org/1999/xhtml">
      The checklist can be processed by varius tools which enable users to:
      <html:ul>
        <html:li>define profiles with customized security policies,</html:li>
        <html:li>audit systems to determine compliance with targeted security policy and</html:li>
        <html:li>adjust system configuration.</html:li>
      </html:ul>
    </html:p>
  </ns0:description>
  <ns0:notice id="disclaimer" xml:lang="en">This XCCDF has been automatically generated, it should only serve as a baseline!</ns0:notice>
  <ns0:front-matter xml:lang="en" />
  <ns0:reference href="http://iase.disa.mil/stigs/index.html">Defense Information Systems Agency - Security Technical Implementation Guides (DISA STIG)</ns0:reference>
  <ns0:reference href="https://www.pcisecuritystandards.org">Payment Card Industry - Data Security Standard (PCI DSS)</ns0:reference>
  <ns0:platform idref="cpe:/o:redhat:enterprise_linux:6" />
  <ns0:version>1.0</ns0:version>
  <ns0:model system="urn:xccdf:scoring:default" />
  <ns0:model system="urn:xccdf:scoring:flat" />
  <ns0:Profile id="xccdf_preupg_profile_default">
    <ns0:title xml:lang="en">Preupgrade assistant default</ns0:title>
    <ns0:description xml:lang="en">
      <html:p xmlns:html="http://www.w3.org/1999/xhtml">
        This profile is designed for the Preupgrade Assistant tool.
      </html:p>
    </ns0:description>
    <ns0:select idref="xccdf_preupg_rule_drivers_ObsoletedNetworkDrivers_obsoletedNetworkDrivers" selected="true" />
    <ns0:select idref="xccdf_preupg_rule_storage_multipath_check" selected="true" />
[ ... snip ... ]
    </ns0:rule-result>
!!!!    <ns0:rule-result idref="xccdf_preupg_rule_usrmgmt_ypserv_service_configs_backup_config" time="2016-10-12T14:45:48" weight="1.000000"> !!!!!
      <ns0:result>notapplicable</ns0:result>
      <ns0:check system="http://open-scap.org/page/SCE">
        <ns0:check-import import-name="stdout">preupg.log.INFO: [unknown-component]: Package  is not installed or it is not signed by Red Hat.
</ns0:check-import>
        <ns0:check-import import-name="stderr">
            </ns0:check-import>
        <ns0:check-export export-name="MODULE_PATH" value-id="xccdf_preupg_value_usrmgmt_ypserv_service_configs_backup_config_state_module_path" />
        <ns0:check-export export-name="RESULT_PART" value-id="xccdf_preupg_value_usrmgmt_ypserv_service_configs_backup_config_state_result_part" />
        <ns0:check-export export-name="CURRENT_DIRECTORY" value-id="xccdf_preupg_value_usrmgmt_ypserv_service_configs_backup_config_state_current_directory" />
        <ns0:check-export export-name="SOLUTION_FILE" value-id="xccdf_preupg_value_usrmgmt_ypserv_service_configs_backup_config_state_solution_file" />
        <ns0:check-export export-name="REPORT_DIR" value-id="xccdf_preupg_value_report_dir" />
        <ns0:check-export export-name="UPGRADE" value-id="xccdf_preupg_value_upgrade" />
        <ns0:check-export export-name="MIGRATE" value-id="xccdf_preupg_value_migrate" />
        <ns0:check-export export-name="DIST_NATIVE" value-id="xccdf_preupg_value_dist_native" />
        <ns0:check-export export-name="DEVEL_MODE" value-id="xccdf_preupg_value_devel_mode" />
        <ns0:check-export export-name="TMP_PREUPGRADE" value-id="xccdf_preupg_value_tmp_preupgrade" />
        <ns0:check-content-ref href="usrmgmt/ypserv/service_configs/backup_config.sh" />
      </ns0:check>
    </ns0:rule-result>
!!!!    <ns0:score maximum="100.000000" system="urn:xccdf:scoring:default">44.430412</ns0:score> !!!!!
!!!!    <ns0:score maximum="69.000000" system="urn:xccdf:scoring:flat">34.000000</ns0:score> !!!!!
  </ns0:TestResult>
</ns0:Benchmark>[root@rhel6 ~]# 

```

**Comment by phracek** [comment]
This depends on #143

**Comment by bocekm** [comment]
New **preupg-diff** tool introduced in pull request https://github.com/upgrades-migrations/preupgrade-assistant/pull/216

**Comment by bocekm** [comment]
prupg-diff tool released in Preupgrade Assistant [v2.3.2](https://github.com/upgrades-migrations/preupgrade-assistant/releases/tag/2.3.2)
